packages/run-protocol/src/vpool-xyk-amm/addPool.js

```diff
@@ -36,48 +74,86 @@ export const makeAddPool = (
     protocolSeat,
   );
 
-  /**
-   * Allows users to add new liquidity pools. `secondaryIssuer` and
-   * its keyword must not have been already used
-   *
-   * @param {ERef<Issuer>} secondaryIssuer
-   * @param {Keyword} keyword - will be used in the
-   * terms.issuers for the contract, but not used otherwise
-   */
-  const addPool = async (secondaryIssuer, keyword) => {
-    const liquidityKeyword = `${keyword}Liquidity`;
-    zcf.assertUniqueKeyword(liquidityKeyword);
+  /** @type {(Brand) => Promise<{poolFacets: PoolFacets, liquidityZcfMint: ZCFMint}>} */
+  const addPool = async secondaryBrand => {
+    const liquidityZcfMint = brandToLiquidityMint.get(secondaryBrand);
 
-    const [secondaryAssetKind, secondaryBrand] = await Promise.all([
-      E(secondaryIssuer).getAssetKind(),
-      E(secondaryIssuer).getBrand(),
-    ]);
+    const { zcfSeat: poolSeat } = zcf.makeEmptySeatKit();
+    /** @type {PoolFacets} */
+    // @ts-expect-error cast
+    const poolFacets = makePool(liquidityZcfMint, poolSeat, secondaryBrand);
+
+    initPool(secondaryBrand, poolFacets);
+    return { liquidityZcfMint, poolFacets };
+  };
+
+  /** @param {ZCFSeat} seat */
+  const addPoolAndLiquidityHandler = async seat => {
+    assertProposalShape(seat, {
+      give: { Central: null, Secondary: null },
+    });
+
+    const {
+      give: { Central: centralAmount, Secondary: secondaryAmount },
+      want: proposalWant,
+    } = seat.getProposal();
+    const secondaryBrand = secondaryAmount.brand;
+
+    const { brand: liquidityBrand, issuer } = brandToLiquidityMint
+      .get(secondaryBrand)
+      .getIssuerRecord();
+
+    const minPoolLiquidity = params.getMinInitialPoolLiquidity();
+
+    if (proposalWant.Liquidity) {
+      const { Liquidity: wantLiquidityAmount } = proposalWant;
+      const liquidityTokensForFunder =
```

Comment:
low risk but this value name reads like it has tokens. consider,
```suggestion
      const centralSurplus =
```

Reply:
how's `centralAboveMinimum`?